load_data.py

```diff
@@ -0,0 +1,115 @@
+from datasets import get_dataset_split_names, load_dataset
+from os import environ
+from tokenizers import Tokenizer
+from tokenizers.models import BPE, Unigram, WordLevel, WordPiece
+from tokenizers.normalizers import NFD, Sequence, Strip, StripAccents
+from tokenizers.trainers import BpeTrainer, UnigramTrainer, WordLevelTrainer, WordPieceTrainer
+from torch.utils.data import DataLoader
+from transformers import PreTrainedTokenizerFast
+
+# Disable parallelism to avoid errors with DataLoaders later on
+environ["TOKENIZERS_PARALLELISM"] = "false"
+
+def get_loaders_tokenizer(dataset_name: str,
+                          tokenizer_name: str,
+                          seq_len: int,
+                          batch_size: int,
+                          vocab_size: int,
+                          data_dir: str,
+                          dataset_config: str=None,
+                          max_token_len: int=20) -> tuple[DataLoader, DataLoader, DataLoader, Tokenizer]:
+    """ Loads the WikiText2 dataset and returns DataLoaders.
+    Args:
+        seq_len (int): Context window/sequence length.
+        batch_size (int): Batch size.
+        vocab_size (int): Maximum vocabulary size.
+
+    Returns:
+        Tuple with the format: (Training DataLoader, Validation DataLoader,
+        Testing DataLoader, Tokenizer object).
+    """
+    assert ["test", "train", "validation"] == sorted(get_dataset_split_names(path=dataset_name, config_name=dataset_config)), \
```

Comment:
Is this meant to be different order between the assert statement and the message?
assert --> test, train, validation
print   --> train, validation, test

Reply:
This code was not ready for review yet, hence the draft PR. The code has now been updated and I'm changing this to a "real" PR, meaning it's now ready for your review. :slightly_smiling_face: